federatedscope/attack/trainer/benign_trainer.py

```diff
@@ -0,0 +1,223 @@
+from calendar import c
+import logging
+from typing import Type
+import torch
+import numpy as np
+
+from federatedscope.core.trainers import GeneralTorchTrainer
+from federatedscope.core.auxiliaries.transform_builder import get_transform
+from federatedscope.attack.auxiliary.backdoor_utils import normalize
+from federatedscope.core.auxiliaries.dataloader_builder import WrapDataset
+from federatedscope.core.auxiliaries.dataloader_builder import get_dataloader
+from federatedscope.core.auxiliaries.ReIterator import ReIterator
+
+# from federatedscope.attack.auxiliary.poisoning_dataset import load_poisoned_dataset_edgeset, load_poisoned_dataset_pixel
+
+logger = logging.getLogger(__name__)
+
+
+
+def wrap_benignTrainer(
+        base_trainer: Type[GeneralTorchTrainer]) -> Type[GeneralTorchTrainer]: 
+    '''
+    Warp the benign trainer for backdoor attack:
+    
+    We just add the normalization operation.
+
+    Args:
+        base_trainer: Type: core.trainers.GeneralTorchTrainer
+
+    :returns:
+        The wrapped trainer; Type: core.trainers.GeneralTorchTrainer
+
+    '''
+
+    # ---- action-level plug-in -------
```

Comment:
It's suggested to remove the unused code.

Reply:
fixed it